src/euphorie/client/browser/session.py

```diff
@@ -792,34 +793,39 @@ def __call__(self):
         return super().__call__()
 
 
-class PublicationMenu(SessionMixin, BrowserView):
-    @property
-    @memoize_contextless
-    def portal(self):
-        """The currently authenticated account."""
-        return api.portal.get()
-
-    def redirect(self):
-        target = self.request.get("HTTP_REFERER") or self.context.absolute_url()
+class PublicationMenu(BrowserView):
+    def redirect(self, target):
         return self.request.response.redirect(target)
 
+    @deprecate("Use the @@locking_view instead. Deprecated in version 15.0.0.dev0")
     def reset_date(self):
         """Reset the session date to now."""
-        session = self.context.session
-        session.published = datetime.now()
-        session.last_publisher = get_current_account()
-        return self.redirect()
+        target = f"{self.context.absolute_url()}/@@locking_view/refresh_lock"
+        return self.redirect(target)
```

Comment:
It looks like you're deprecating the feature and changing it at the same time. Isn't deprecation meant to give someone who uses the feature time to adapt before it is changed or removed?

Reply:
I added the deprecation to the UI, allowing to still use it with the updated backend.
There is an upgrade step that updates the data in such a way that this deprecated UI is still functional but behaves like an alias for the new logic

Reply:
Not sure about it but let's leave it like this.